### PR TITLE
feat: implement `talosctl config info` command

### LINKED ISSUE #3852: Quality of life improvements for RBAC
- [x] Implement `talosctl config info` command that shows information about roles for the current context.
- [x] Add information about enabled features to Version API and `talosctl version` command output.
- [x] Make e2e tests pass.

### PR DESCRIPTION
**Pull request by AlekSi**
Closes #3852.

Signed-off-by: Alexey Palazhchenko <alexey.palazhchenko@gmail.com>

### CODE REVIEW

**Review by smira — COMMENTED** [review]
lgtm, it might be nice to add certificate validity time (or expiration)

File: cmd/talosctl/cmd/talos/config_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        crt: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJNekNCNXFBREFnRUNBaEFQL1MrVGx0WTBHdGk5Q1g0UDNVZStNQVVHQXl0bGNEQVFNUTR3REFZRFZRUUsKRXdWMFlXeHZjekFlRncweU1UQTNNRFV4TVRRek1qRmFGdzB6TVRBM01ETXhNVFF6TWpGYU1CUXhFakFRQmdOVgpCQW9UQ1c5ek9tWjFkSFZ5WlRBcU1BVUdBeXRsY0FNaEFLck04NmtPYm1MdGw5OVdpdzFFL29pdnl2YXVqVmNkCmlQTk82TVhQNGxEMm8xSXdVREFPQmdOVkhROEJBZjhFQkFNQ0I0QXdIUVlEVlIwbEJCWXdGQVlJS3dZQkJRVUgKQXdFR0NDc0dBUVVGQndNQ01COEdBMVVkSXdRWU1CYUFGSEs3MXJqNU5ObFp3UnVWV3BXOG5DWm4rdmFsTUFVRwpBeXRsY0FOQkFDSno3NTlGeVFJMXlIWTJNRG9vZDNrZjdZeG9HRG1Hem1nNllqRHJueXAxWGpaQ3o1Q1RQbm9jCjhxWlAyTXE5MDJnOXZSSUh1dm84N0NIZjJacTNGZ1U9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
+        key: LS0tLS1CRUdJTiBFRDI1NTE5IFBSSVZBVEUgS0VZLS0tLS0KTUM0Q0FRQXdCUVlESzJWd0JDSUVJS1RCRDIyZDBLVnNTek5iSkNBdjNObUVnL1VWOTk4SHZvY2NGb1lDOEJ1bAotLS0tLUVORCBFRDI1NTE5IFBSSVZBVEUgS0VZLS0tLS0K
+`,
+			expected: strings.TrimSpace(`
+Current context:     future-role
+Nodes:               not defined
+Endpoints:           not defined
+Roles:               os:future
+Certificate expires: 10 years from now (2031-07-03)
```
Comment (smira) [inline cmd/talosctl/cmd/talos/config_test.go:88]: do we need to update this value once a year on 3rd of July? :)

**Review by smira — APPROVED** [review]

File: cmd/talosctl/cmd/talos/config_test.go (hunk 1 above)
Comment (AlekSi) [reply cmd/talosctl/cmd/talos/config_test.go:88]: No, `now` is fixed in tests: https://github.com/talos-systems/talos/pull/3888/files#diff-0db14a18214a13d70974f2e1f0b7e280c81d37764922d595a8198aad178c71aaR21

**Pull request merged**